src/utils/misc.ts

```diff
@@ -50,3 +50,19 @@ export function loadRentalOffer(rentalOfferId: string): RentalOffer {
 
   return rentalOffer;
 }
+
+export function updateLastOfferExpirationAt(nft: Nft, expirationDate: BigInt): void {
+  if (nft.lastOfferExpirationAt && nft.lastOfferExpirationAt!.gt(expirationDate)) return
+  nft.lastOfferExpirationAt = expirationDate
+  nft.save()
+}
+
+export function removeLastOfferExpirationAt(nftId: string): void {
```

Comment:
This function is called whenever an Offer is accepted or canceled. However, it's important to note that the offer in question might not be the one with the highest expiration date.

We need to pass the expiration date of the Offer to this function and only remove it if it's equal to or higher the `nft.lastOfferExpirationAt`